### PR TITLE
[Bouffalo Lab] Update gcc toolchain and flash tool of docker image

### PR DESCRIPTION
**Pull request by wy-hh**
- Update Bouffalo Lab docker image
  - update flash tool: use BouffaloLabDevCube-v1.9.0 instead of bflb-iot-tool
  - update thead gcc toolchain download url.

### CODE REVIEW

File: .github/workflows/examples-bouffalolab.yaml [hunk 1]
```diff
@@ -37,7 +37,7 @@ jobs:
     if: github.actor != 'restyled-io[bot]'
 
     container:
-      image: ghcr.io/project-chip/chip-build-bouffalolab:54
+      image: ghcr.io/project-chip/chip-build-bouffalolab:67
```
Comment (andy31415) [inline .github/workflows/examples-bouffalolab.yaml:40]: This change has to be made in a separate PR from the dockerfile update: we update the dockerfile after the commit, so there ia few hours lag until they are available.

Comment (wy-hh) [reply .github/workflows/examples-bouffalolab.yaml]: Updated to only contain chagnes for Docker image build.
Rest changes will be in another PR when Docker image is accessible.

**Review by woody-apple — APPROVED** [review]

**Review by andy31415 — APPROVED** [review]
